### PR TITLE
MatrixSymbol, support expand

### PR DESCRIPTION
**Pull request by vruge**

### CODE REVIEW

**Comment by vruge** [comment]
How I can fix

```bash
/home/travis/build/sympy/sympy/sympy/matrices/expressions/matmul.py:docstring
 of sympy.matrices.expressions.MatMul.expand:0:Inline emphasis start-string without end-string.
```

**Comment by jksuom** [comment]
`A *(B + C) = A*B + A*C` should probably be enclosed between a pair of double backquotes.

**Comment by vruge** [comment]
https://travis-ci.org/sympy/sympy/jobs/291099932 , failed agian for html

File: sympy/matrices/expressions/matmul.py [hunk 1]
```diff
@@ -120,6 +120,37 @@ def doit(self, **kwargs):
             args = self.args
         return canonicalize(MatMul(*args))
 
+
+    def expand(self, *args, **kwargs):
+        #expand MatMul
+        # e.g.
+        # A *(B + C) = A*B + A*C
+        # (B+C)*A = B*A + B*C
+        # expand((A+D)*(B + C), deep=False) = (A+D) *B + (A+D)*C
```
Comment (smichr) [inline sympy/matrices/expressions/matmul.py]: I don't believe this matches the response when dealing with Expr...deep refers to traversing arguments of functions.

**Comment by gxyd** [comment]
Can you please resolve merge conflicts here? I'll have a look then.

**Comment by vruge** [comment]
done

**Comment by czgdp1807** [comment]
@jksuom Please provide status update on this PR.

**Comment by czgdp1807** [comment]
@vruge Are you still interested to work on it?

**Comment by jksuom** [comment]
Problems of this type are discussed in https://github.com/sympy/sympy/issues/18181. It looks like there might be different ways of implementing `expand` on matrix-like objects.

**Comment by oscarbenjamin** [comment]
Matrix expressions subclass Expr so I don't think they need #18181.

Is the problem that this PR solves already fixed now?

```julia
In [10]: A, B, C = [MatrixSymbol(n, 2, 2) for n in 'ABC']                                                                                      

In [11]: expr = A*(B+C)                                                                                                                        

In [12]: expr                                                                                                                                  
Out[12]: A⋅(B + C)

In [13]: expr.expand()                                                                                                                         
Out[13]: A⋅B + A⋅C

In [14]: expand(expr)                                                                                                                          
Out[14]: A⋅B + A⋅C
```

Ah I see it doesn't work for `MatPow`:
```julia
In [18]: expand((A + B)**2)                                                                                                                    
Out[18]: 
       2
(A + B) 
```
I don't think this PR fixes that though.

**Comment by czgdp1807** [comment]
Closing in reference to the above comment, in addition, the repository from which this PR was made has been deleted. Please let me know if there is a need to raise issue, for `MatPow`. Thanks for the contributions.

**Pull request closed without merging**